src/Features/LanguageServer/ProtocolUnitTests/Configuration/DidChangeConfigurationNotificationHandlerTest.cs

```diff
@@ -0,0 +1,267 @@
+            // Let client has a default values for all options.
+            clientCallbackTarget.SetClientSideOptionValues(setToDefaultValue: true);
+
+            await server.ExecuteRequestAsync<DidChangeConfigurationParams, object>(Methods.WorkspaceDidChangeConfigurationName, new DidChangeConfigurationParams(), CancellationToken.None).ConfigureAwait(false);
+            await WaitAsync(server.TestWorkspace, Methods.WorkspaceDidChangeConfigurationName).ConfigureAwait(false);
+            VerifyValuesInServer(server.TestWorkspace, clientCallbackTarget.MockClientSideValues);
+        }
+
+        private static void VerifyValuesInServer(TestWorkspace workspace, List<string> expectedValues)
+        {
+            var globalOptionService = workspace.GetService<IGlobalOptionService>();
+            var supportedOptions = DidChangeConfigurationNotificationHandler.SupportedOptions;
+            Assert.Equal(supportedOptions.Sum(option => option is IPerLanguageValuedOption ? 2 : 1), expectedValues.Count);
+            var optionsAndLanguageToVerify = supportedOptions.SelectManyAsArray(option => option is IPerLanguageValuedOption
+                ? DidChangeConfigurationNotificationHandler.SupportedLanguages.SelectAsArray(lang => (option, lang))
+                : SpecializedCollections.SingletonEnumerable((option, string.Empty)));
+
+            for (var i = 0; i < expectedValues.Count; i++)
+            {
+                var (option, languageName) = optionsAndLanguageToVerify[i];
+                var valueFromClient = expectedValues[i];
+
+                Assert.True(option.Definition.Serializer.TryParse(valueFromClient, out var result));
+                if (option is IPerLanguageValuedOption)
+                {
+                    var valueInServer = globalOptionService.GetOption<object>(new OptionKey2(option, languageName));
+                    Assert.Equal(result, valueInServer);
+                }
+                else
+                {
+                    var valueInServer = globalOptionService.GetOption<object>(new OptionKey2(option, null));
+                    Assert.Equal(result, valueInServer);
+                }
+            }
+        }
+
+        private static async Task WaitAsync(TestWorkspace testWorkspace, string name)
+        {
+            var listenerProvider = testWorkspace.GetService<AsynchronousOperationListenerProvider>();
+            await listenerProvider.GetWaiter(name).ExpeditedWaitAsync();
+        }
+
+        private class ClientCallbackTarget
+        {
+            public bool WorkspaceDidChangeConfigurationRegistered { get; private set; } = false;
+            public List<ConfigurationItem> ReceivedConfigurationItems { get; } = new();
+            public List<string> MockClientSideValues { get; } = new();
+
+            [JsonRpcMethod(Methods.ClientRegisterCapabilityName)]
+            public void ClientRegisterCapability(JToken @params, CancellationToken _)
```

Comment:
Can this be RegistrationParams instead?  You might need to set the property UseSingleArgumentDeserialization (forget the exact name) in the [JsonRpcMethod(...)] attribute to use it